src/huntsman/pocs/observatory.py

```diff
@@ -310,66 +320,79 @@ def take_dark_fields(self,
         if not isinstance(n_darks, list):
             n_darks = listify(n_darks) * len(exptimes)
 
+        if set_from_config:
+            dark_config = self.get_config('calibration', default=dict())
+            exptimes = dark_config["darks"]["exposure_time"]
+            n_darks = dark_config["darks"]["n_darks"]
+
         # Loop over cameras.
-        for exptime, num_darks in zip(exptimes, n_darks):
+        if len(exptimes) > 0:
+            self.logger.info(f"Going to take {n_darks} darks for each of these exposure times {exptimes}")
 
-            start_time = current_time()
+            for exptime, num_darks in zip(exptimes, n_darks):
+
+                start_time = current_time()
 
-            with suppress(AttributeError):
-                exptime = exptime.to_value(u.second)
+                with suppress(AttributeError):
+                    exptime = exptime.to_value(u.second)
 
-            dark_obs = self._create_dark_observation(exptime)
+                # Create dark observation
+                dark_obs = self._create_dark_observation(exptime)
 
-            # Loop over exposure times for each camera.
-            for num in range(num_darks):
+                # Loop over exposure times for each camera.
+                for num in range(num_darks):
 
-                self.logger.debug(f'Darks sequence #{num} of exposure time {exptime}s')
+                    self.logger.debug(f'Darks sequence #{num} of exposure time {exptime}s')
 
-                camera_events = dict()
+                    camera_events = dict()
 
-                # Take a given number of exposures for each exposure time.
-                for camera in cameras_list.values():
-                    # Create dark observation
-                    fits_headers = self.get_standard_headers(observation=dark_obs)
-                    # Common start time for cameras
-                    fits_headers['start_time'] = flatten_time(start_time)
+                    # Take a given number of exposures for each exposure time.
+                    for camera in cameras_list.values():
 
-                    # Create filename
-                    path = os.path.join(image_dir,
-                                        'darks',
-                                        camera.uid,
-                                        dark_obs.seq_time)
+                        # Set header
+                        fits_headers = self.get_standard_headers(observation=dark_obs)
+                        # Common start time for cameras
+                        fits_headers['start_time'] = flatten_time(start_time)
 
-                    filename = os.path.join(
-                        path, f'{imtype}_{num:02d}.{camera.file_extension}')
+                        # Create filename
+                        path = os.path.join(dark_obs.directory,
```

Comment:
I think this can fit all on one line. The line limit for `panoptes` projects is 100 characters. Same for some of the other lines in this function.